File: jumbotravel-api/domain/dto/agent_flight.go
package dto

import "time"

type AgentFlight struct {
	RouteID             *int       `json:"route_id"`
	DepartureCountry    *string    `json:"departure_country"`
	DepartureCity       *string    `json:"departure_city"`
	ArrivalCountry      *string    `json:"arrival_country"`
	ArrivalCity         *string    `json:"arrival_city"`
	DepartureAirport    *string    `json:"departure_airport"`
	ArrivalAirport      *string    `json:"arrival_airport"`
	DepartureCommonName *string    `json:"departure_commonname"`
	ArrivalCommonName   *string    `json:"arrival_commonname"`
	AirplaneID          *int       `json:"airplane_id"`
	Carrier             *string    `json:"carrier"`
	FlightNumber        *string    `json:"flight_number"`
	Seats               *int       `json:"seats"`
	FlightID            *int       `json:"flight_id"`
	Status              *string    `json:"status"`
	DepartureTime       *time.Time `json:"departure_time"`
	ArrivalTime         *time.Time `json:"arrival_time"`
	FlightCreated       *time.Time `json:"flight_created"`
	FlightUpdated       *time.Time `json:"flight_updated"`
	UpdatedAt           *time.Time `json:"updated_at"`
	CreatedAt           *time.Time `json:"created_at"`
} // @name AgentFlight

func (v *AgentFlight) GetDestFields() []interface{} {
	return []interface{}{
		&v.RouteID,
		&v.AirplaneID,
		&v.Carrier,
		&v.FlightNumber,
		&v.Seats,
		&v.FlightID,
		&v.Status,
		&v.DepartureTime,
		&v.ArrivalTime,
		&v.FlightCreated,
		&v.FlightUpdated,
		&v.DepartureCountry,
		&v.ArrivalCountry,
		&v.DepartureCity,
		&v.ArrivalCity,
		&v.DepartureAirport,
		&v.ArrivalAirport,
		&v.DepartureCommonName,
		&v.ArrivalCommonName,
		&v.UpdatedAt,
		&v.CreatedAt,
	}
}

func (v *AgentFlight) New() {
	*v = AgentFlight{}
}

func (v *AgentFlight) Val() interface{} {
	return *v
}

type FlightAgent struct {
	AgentID *int    `json:"agent_id"`
	Name    *string `json:"name"`
	Surname *string `json:"surname"`
	Email   *string `json:"email"`
} // @name FlightAgent

func (v *FlightAgent) GetDestFields() []interface{} {
	return []interface{}{
		&v.AgentID,
		&v.Name,
		&v.Surname,
		&v.Email,
	}
}

func (v *FlightAgent) New() {
	*v = FlightAgent{}
}

func (v *FlightAgent) Val() interface{} {
	return *v
}

File: jumbotravel-portal/src/api/domain/agent_flight.js
class Flight {

    constructor({
        route_id,
        departure_country,
        departure_city,
        arrival_country,
        arrival_city,
        departure_airport,
        arrival_airport,
        departure_commonname,
        arrival_commonname,
        airplane_id,
        carrier,
        flight_number,
        seats,
        flight_id,
        status,
        departure_time,
        arrival_time,
        flight_created,
        flight_updated,
        updated_at,
        created_at,
        has_booking,
    }) {
        this.route_id = route_id;
        this.departure_country = departure_country;
        this.departure_city = departure_city;
        this.arrival_country = arrival_country;
        this.arrival_city = arrival_city;
        this.departure_airport = departure_airport;
        this.arrival_airport = arrival_airport;
        this.departure_commonname = departure_commonname;
        this.arrival_commonname = arrival_commonname;
        this.airplane_id = airplane_id;
        this.carrier = carrier;
        this.flight_number = flight_number;
        this.seats = seats;
        this.flight_id = flight_id;
        this.status = status;
        this.departure_time = departure_time;
        this.arrival_time = arrival_time;
        this.flight_created = flight_created;
        this.flight_updated = flight_updated;
        this.updated_at = updated_at;
        this.created_at = created_at;
        this.has_booking = has_booking;
    }

    getRoute() {
        return {
            "departure": this.departure_airport,
            "arrival": this.arrival_airport,
        }
    }

}

export default Flight;